sdk/tests/conformance2/reading/read-pixels-pack-parameters.html

```diff
@@ -115,12 +115,22 @@
         var array = new Uint8Array(arrBuffer);
     }
 
-    // Check the last pixel of the last row.
-    var bytesPerRow = actualWidth * bytesPerPixel + padding;
-    var pos = skipSize + bytesPerRow * (height - 1) + (width - 1) * bytesPerPixel;
-    for (var i = 0; i < bytesPerPixel; ++i)
-        pixel[i] = array[pos + i];
-    shouldBe("pixel", "expectedColor");
+    // Check the first and last pixels of each row.
+    // TODO(zmo): Revisit this if row_length < width cases are allowed.
+    for (var row = 0; row < height; ++row) {
+        var bytesPerRow = actualWidth * bytesPerPixel + padding;
+        // First pixel
+        var pos = skipSize + bytesPerRow * row;
+        for (var i = 0; i < bytesPerPixel; ++i)
+            pixel[i] = array[pos + i];
+        shouldBe("pixel", "expectedColor");
```

Comment:
Sure.  Currently we use clear() to set the fbo color, we just need to switch to shaders and draw()